### PR TITLE
Remove farms public IP delete button and create  delete all action with it's module and method

### PR DESCRIPTION
**Pull request by samaradel**
### Description

- Remove the delete button and add delete all action.
- Create ```removeFarmIps``` module.
- Create ```removeFarmIps``` method in tf-grid.
- Document ```removeFarmIps``` methods.
### Changes

[Screencast from 21-04-24 17:17:26.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/11662805/c7372867-f2a2-41c2-8a4d-09d00512c2c1)

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2561
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2610

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by Mahmoud-Emad** [comment]
Does the loader color want to be red?
image

File: packages/playground/src/dashboard/components/public_ips_table.vue [hunk 1]
```diff
@@ -163,19 +169,24 @@ export default {
...
-        createCustomToast("IP is deleted successfully!", ToastType.success);
-        publicIps.value.splice(index, 1);
+        for (const item of itemsToDelete.value) {
+          await gridStore.grid.farms.removeFarmIp({
+            ip: item.ip,
+            farmId: props.farmId,
+          });
+          createCustomToast("IP is deleted successfully!", ToastType.success);
+          await getFarmByID(props.farmId);
+        }
```
Comment (Mahmoud-Emad) [inline packages/playground/src/dashboard/components/public_ips_table.vue:182]: When considering a list of more than 20 IP addresses, displaying them all in a toast notification might overwhelm the user. Instead, it's better to consolidate the information and display only one concise message after removing all the IPs. What are your thoughts on this approach?

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]
Good job ya Samar!

File: packages/playground/src/dashboard/components/public_ips_table.vue (hunk 1 above)
Comment (samaradel) [reply packages/playground/src/dashboard/components/public_ips_table.vue]: totally agree

File: packages/playground/src/dashboard/components/public_ips_table.vue [hunk 2]
```diff
@@ -163,19 +185,24 @@ export default {
...
         isRemoving.value = true;
-        await gridStore.grid.farms.removeFarmIp({ ip: options.ip, farmId: options.farmId });
-        createCustomToast("IP is deleted successfully!", ToastType.success);
-        publicIps.value.splice(index, 1);
+        for (const item of itemsToDelete.value) {
+          await gridStore.grid.farms.removeFarmIp({
+            ip: item.ip,
+            farmId: props.farmId,
+          });
+        }
```
Comment (0oM4R) [inline packages/playground/src/dashboard/components/public_ips_table.vue:197]: this makes things take time, we can add an method to grid client that uses batchall to delete all of them in one request

**Review by 0oM4R — CHANGES_REQUESTED** [review]

File: packages/tfchain_client/src/farms.ts [hunk 3]
```diff
@@ -89,7 +90,16 @@ class Farms extends QueryFarms {
...
+    return this.client.patchExtrinsic<void>(extrinsic);
+  }
+
+  @checkConnection
+  async removeFarmIps(options: RemoveFarmIPOptions[]) {
+    const extrinsics: ExtrinsicResult<void>[] = [];
+    for (const option of options) {
+      extrinsics.push(await this.removeFarmIp(option));
+    }
+    await this.client.applyAllExtrinsics<void>(extrinsics);
```
Comment (AhmedHanafy725) [inline packages/tfchain_client/src/farms.ts:102]: this method shouldn't be added to the chain client, the optimal place is [here](https://github.com/threefoldtech/tfgrid-sdk-ts/tree/development/packages/grid_client/src/clients/tf-grid). please add a new file for the farms and implement it there.

this is an [example](https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/grid_client/src/clients/tf-grid/contracts.ts#L319-L326) for deleting the contracts with a batch call.

Comment (samaradel) [reply packages/tfchain_client/src/farms.ts]: function created

File: packages/grid_client/src/clients/tf-grid/farms.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+class TFFarms extends Farms {
+  async removeFarmIps(options: RemoveFarmIPModel[]) {
+    const extrinsics: ExtrinsicResult<Farm>[] = [];
+    for (const option of options) {
+      extrinsics.push(await this.removeFarmIp(option));
+    }
+    await this.client.applyAllExtrinsics(extrinsics);
+  }
+}
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/clients/tf-grid/farms.ts:13]: Let's document the class and the method

File: packages/grid_client/src/modules/farms.ts [hunk 5]
```diff
@@ -39,6 +39,13 @@ class Farms {
     return (await this.client.farms.removeFarmIp(options)).apply();
   }
 
+  @expose
+  @validateInput
+  @checkBalance
+  async removeFarmIps(options: RemoveFarmIPModel[]) {
+    return await this.client.farms.removeFarmIps(options);
+  }
+
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/modules/farms.ts:48]: Here also, please document it

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]
Good job :)

**Review by Mahmoud-Emad — APPROVED** [review]
Good job ya Samar

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**